extra/resources/ifspeed

```diff
@@ -409,20 +450,28 @@ bond_get_speed() {
         bond_get_active_iface ${iface} active_iface
         aggregate_speed=$( iface_get_speed $active_iface )
     fi
-    if [ -n "$2" ] ; then # Record value in specified var. This expects we were called not in a sub-shell. 
+    if [ -n "$2" ] ; then # Record value in specified var. This expects we were called not in a sub-shell.
         eval $2=\${aggregate_speed}
     else # Expect sub-shell
         echo ${aggregate_speed}
     fi
 }
 
 update() {
-    local speed=$( iface_get_speed ${OCF_RESKEY_iface} )
+    local nic=${OCF_RESKEY_iface};
+    if [ -z "${OCF_RESKEY_iface}" ]; then
+        nic=`get_nic_name_by_ip`
+        if [ -z "${nic}" ];then
+            ocf_log err "Could not retrieve network interface name from ip address (${OCF_RESKEY_ip})"
+            exit $OCF_ERR_GENERIC
+        fi
+    fi
+    local speed=$( iface_get_speed ${nic} )
```

Comment:
Sorry for not spotting this earlier, could you please move 'local' declaration back to the beginning of the function?
Yes, I understand, that doesn't make any difference in shell, but I came from C, and such a style is preserved over the whole RA.
What I mean:
<pre>update() {
    local speed
    [... your code ...]
    speed=$( iface_get_speed ${nic} )
</pre>
Thanks for understanding.

Reply:
I will change that.